### PR TITLE
Fix #1823 xy plot working fine in oscilloscope

### LINKED ISSUE #1823: XY plot feature doesn't work properly in oscilloscope
**Actual Behaviour**
XY plot feature doesn't work properly in oscilloscope

**Expected Behaviour**
Implement the functionality similar to desktop app

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1823

**Changes**: Fixed xyPlot code in oscilloscope

**Screenshot/s for the changes**: 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 

[osc_xy_plot3.zip](https://github.com/fossasia/pslab-android/files/3329397/osc_xy_plot3.zip)

### CODE REVIEW

File: app/src/main/java/io/pslab/activity/OscilloscopeActivity.java [hunk 1]
```diff
@@ -441,12 +441,10 @@ public void run() {
                         }
                     }
 
-                    if (scienceLab.isConnected() && viewIsClicked && isXYPlotSelected) {
+                    if (scienceLab.isConnected() && isXYPlotSelected) {
+                        Log.d("xy plot", "called");
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Let's remove this part.

File: app/src/main/java/io/pslab/activity/OscilloscopeActivity.java [hunk 2]
```diff
@@ -1092,69 +1090,34 @@ protected void onPostExecute(Void aVoid) {
...
+        private String analogInput2;
+        private float[] xFloatData;
+        private float[] yFloatData;
 
         @Override
         protected Void doInBackground(String... params) {
-            HashMap<String, double[]> data;
-            if ("CH2".equals(xyPlotXAxisChannel) || "CH2".equals(xyPlotYAxisChannel)) {
-                analogInput = params[0];
-                data = scienceLab.captureTwo(1000, 10, analogInput, false);
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Ideal method would be this. With current implementation it fetches one trace and then fetch the second trace after sometime; not simultaneous. CaptureTwo fetches them simultaneously.

File: app/src/main/java/io/pslab/activity/OscilloscopeActivity.java (hunk 1 above)
Comment (neel1998) [reply app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Yes. I added for debugging

File: app/src/main/java/io/pslab/activity/OscilloscopeActivity.java (hunk 2 above)
Comment (neel1998) [reply app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Okay. will make the change

**Comment by neel1998** [comment]
@CloudyPadmal  the ```captureTwo()```function is giving weird output. Also i couldn't understand how ```captureTwo()``` can be used to get data for two random channels say CH3 and MIC. I also checked the code of new desktop App , there also ```fetchTrace()``` function is called sequentially. For the delay you mentioned earlier i have remove the sleep() call between two ```fetchTrace()``` calls

**Comment by CloudyPadmal** [comment]
> Also i couldn't understand how `captureTwo()` can be used to get data for two random channels say CH3 and MIC.

I looked through the code for you because you couldn't understand. If you compare the changes you made with the previous code, it's already there anyways. 
- [`CaptureTwo`](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/communication/ScienceLab.java#L533) fetches `CH1` and `CH2`. If you want random channel selection, it was in an else statement in previous code using [`CaptureFour`](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/communication/ScienceLab.java#L533) where it fetches all four channels `CH1`, `CH2`, `CH3`, `MIC`. From that you can filter out the channels you want.
- May be it's a good thing if you can include JavaDocs to those methods as you read and understand.

If the channel readings are giving weird outputs, may be increase timeout/sleep periods in methods in [`ScienceLab.java`](https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/communication/ScienceLab.java) class.

> I also checked the code of new desktop App , there also `fetchTrace()` function is called sequentially.

I don't remember how many times I've said it, take only `Python Desktop App` as reference for code implementations (Not for UI). I'll open an issue in Electron repo for the case.

**Comment by neel1998** [comment]
@CloudyPadmal thanks for your help and time. Will keep your advice in mind regarding desktop app. Sorry for inconvenience caused

**Comment by neel1998** [comment]
@CloudyPadmal  have updated the apk and gif. Have a look. Thanks

**Pull request merged**